### PR TITLE
Add ADAr (de-)serialisation

### PR DESCRIPTION
**Pull request by SkyToGround**
As the title says. Also some minor updates to the documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,23 +6,22 @@ https://github.com/ess-dmsc/streaming-data-types
...
+|f142|Log data|
+|ev42|Event data|
+|x5f2|Status messages|
+|tdct|Timestamps|
+|ep00|EPICS connection info|
+|rf5k|Forwarder configuration update|
+|answ|File-writer command response|
+|wrdn|File-writer finished writing|
+|NDAr|**Deprecated**|
+|ADAr|For storing EPICS areaDetector data|
```
Comment (mattclarke) [inline README.md:24]: Thanks, I keep meaning to do this

File: streaming_data_types/area_detector_ADAr.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        source: str,
+        data: Union[np.ndarray, str, int, float],
+    ):
+        self.name = name
+        self.description = description
+        self.source = source
+        self.data = data
+
+    def __eq__(self, other):
+        data_is_equal = type(self.data) == type(other.data)  # noqa: E721
```
Comment (mattclarke) [inline streaming_data_types/area_detector_ADAr.py:28]: isinstance will make the warning go away (and look nicer)

Comment (mattclarke) [reply streaming_data_types/area_detector_ADAr.py:28]: probably

Comment (SkyToGround) [reply streaming_data_types/area_detector_ADAr.py:28]: `isinstance()` does not do what I want to do here.

Comment (mattclarke) [reply streaming_data_types/area_detector_ADAr.py:28]: Fair enough

**Review by mattclarke — APPROVED** [review]
LGTM

**Pull request merged**